tests/engine/test_perturb.py

```diff
@@ -20,106 +17,66 @@
-        )
-        self.assertAlmostEqual(
-            mean_diff1,
-            AMPLITUDE * 0.5,
-            msg="perturbation is most likely too small!",
-            delta=AMPLITUDE * 1e-2,
-        )
-        self.assertAlmostEqual(
-            mean_diff2,
-            AMPLITUDE * 0.5,
-            msg="perturbation did not return a copy of input!",
-            delta=AMPLITUDE * 1e-2,
-        )
-
-    def test_perturb_nc(self):
-
-        # create two dummy netcdf4 files, one with single, one with double precision
-        for i, dt in enumerate([np.float32, np.float64]):
-            self.data[i] = Dataset(f"dummy{i}.nc", "w")
-            self.data[i].createDimension("x", size=ARRAY_DIM)
-            self.data[i].createDimension("y", size=ARRAY_DIM)
-            self.data[i].createVariable("z", dt, dimensions=("x", "y"))
-            self.data[i].variables["z"][:] = np.ones((ARRAY_DIM, ARRAY_DIM))
-
-        # perturb the double precision file
-        zd = self.data[1].variables["z"][:]
-        zd_perturb = perturb_array(zd, 10, AMPLITUDE)
-        self.data[1].variables["z"][:] = zd_perturb
-
-        # close the data
-        for i in range(2):
-            self.data[i].close()
-
-        # reopen the files to make sure we get the values form disk
-        for i in range(2):
-            self.data[i] = Dataset(f"dummy{i}.nc", "r")
-        zf = self.data[0].variables["z"][:]
-        zd_perturb = self.data[1].variables["z"][:]
-
-        # compute some stats and do assertions
-        diff = zf - zd_perturb
-        mean_diff = np.mean(np.abs(diff))
-        self.assertTrue(zf.dtype == "float32", msg="zf is not float32!")
-        self.assertTrue(zd.dtype == "float64", msg="zf is not float64!")
-        self.assertLess(
-            np.max(diff), AMPLITUDE, msg="perturbation is larger than amplitude!"
-        )
-        self.assertAlmostEqual(
-            mean_diff,
-            AMPLITUDE * 0.5,
-            msg="perturbation is most likely too small!",
-            delta=AMPLITUDE * 1e-2,
-        )
-
-        # to really make sure:
-        # create a plot of the difference to be read out manually if you are scared
-        fig, ax = plt.subplots(1, 1)
-        xx, yy = np.meshgrid(np.linspace(0, 1, ARRAY_DIM), np.linspace(0, 1, ARRAY_DIM))
-        cs = ax.contourf(xx, yy, diff)
-        fig.colorbar(cs, ax=ax)
-        fig.savefig(f"{self.test_path}/diff_figure.pdf")
-
-
-if __name__ == "__main__":
-    unittest.main()
+@pytest.fixture(name="create_nc_files")
+def fixture_create_nc_files():
+    data = [None] * 2
+    for i, dt in enumerate([np.float32, np.float64]):
+        data[i] = Dataset(f"dummy{i}.nc", "w")
+        data[i].createDimension("x", size=ARRAY_DIM)
+        data[i].createDimension("y", size=ARRAY_DIM)
+        data[i].createVariable("z", dt, dimensions=("x", "y"))
+        data[i].variables["z"][:] = np.ones((ARRAY_DIM, ARRAY_DIM))
+    yield data
+    for i in range(2):
+        data[i].close()
```

Comment:
This netcdf-files are not written to disk right?
Simply kept in memory?

Reply:
I am not sure, I would assume when you close them they get written?

Reply:
Ok, just added tmp_dir here.

Reply:
Then it is fine, tmp-dir is whiped out at the end of the test session.